test/functional/web/safari-nativewebtap-e2e-specs.js

```diff
@@ -44,22 +55,47 @@ describe('Safari - coordinate conversion -', function () {
     // Close all tabs in Safari before running tests because left over tabs can affect the test results.
     // If possible, it's probably better to kick com.apple.mobilesafari.settings.DeleteAllDataAndCachesTask
     // task in com.apple.Preferences app or something equivalent rather than closing tabs via GUI.
-    async function closeAllTabsViaSettingsApp (deviceName) {
+    async function closeAllTabsViaSettingsApp(deviceName) {
       const newCaps = {
         'appium:deviceName': deviceName,
         'appium:usePrebuiltWDA': hasDefaultPrebuiltWDA(),
       };
       const localSettingsCaps = amendCapabilities(SETTINGS_CAPS, newCaps);
       const driver = await initSession(localSettingsCaps);
-      await driver.$(CLASS_CHAIN_SEARCH + ':**/XCUIElementTypeStaticText[`label == "Safari"`]').click();
+      await driver
+        .$(CLASS_CHAIN_SEARCH + ':**/XCUIElementTypeStaticText[`label == "Safari"`]')
+        .click();
       await driver.$('~CLEAR_HISTORY_AND_DATA').click();
-      if (await driver.$$('~Clear').length > 0) { // for iPad
-        await driver.$('~Clear').click();
-      } else { // for iPhone
-        await driver.$('~Clear History and Data').click();
-      }
-      if (util.compareVersions(extractCapabilityValue(localSettingsCaps, 'appium:platformVersion'), '>=', '16.0')) {
-        await driver.$(CLASS_CHAIN_SEARCH + ':**/XCUIElementTypeButton[`label == "Close Tabs"`]').click();
+
+      if (
+        util.compareVersions(
+          extractCapabilityValue(localSettingsCaps, 'appium:platformVersion'),
+          '>=',
+          '17.0'
+        )
+      ) {
+        // TODO: please clear all tabs as well if needed.
```

Comment:
To fix current failing tests, maybe here should clear tabs as well. Let me update